modules/display.c

```diff
@@ -3575,8 +3575,8 @@ static void mdy_brightness_set_lpm_level(gint level)
 
 static void mdy_brightness_set_on_level(gint hbm_and_level)
 {
-    gint new_brightness = (hbm_and_level >> 0) & 0xff;
-    gint new_hbm_level  = (hbm_and_level >> 8) & 0xff;
+    gint new_brightness = hbm_and_level;
+    gint new_hbm_level  = hbm_and_level == 100;
```

Comment:
This change goes against design (excerpt from datapipe.c)

    Display brightness;
    bits 0-7 is brightness in percent (0-100)
    upper 8 bits is high brightness boost (0-2)

AFAIK, IIRC, etc the idea is basically that

1. hbm is triggered by automatic brightness tuning
2. duration is limited even if conditions persist
3. you can't choose hbm from settings (i.e. having 100 trigger hbm is not ok)

So, instead of code change like this you should have a custom als configuration file where high lux values yield brightness value like 356 instead of 100 and install it as e.g.  "/etc/mce/60-als-devicename.ini" so that it overrides defaults originating from /etc/mce/20als-defaults.ini.